crates/hyperqueue/src/dashboard/ui/screens/home/screen.rs

```diff
@@ -1,34 +1,39 @@
 use termion::event::Key;
-use tui::layout::{Constraint, Direction, Rect};
 
 use crate::dashboard::ui::screen::{ClusterState, Screen};
+use crate::dashboard::ui::screens::home::chart_section::ClusterOverviewChart;
 use crate::dashboard::ui::screens::home::info_section::WorkerInfoTable;
 use crate::dashboard::ui::screens::home::worker_utilization_table::WorkerUtilTable;
 use crate::dashboard::ui::styles::style_header_text;
 use crate::dashboard::ui::terminal::DashboardFrame;
 use crate::dashboard::ui::widgets::text::draw_text;
 
+use tui::layout::{Constraint, Direction, Layout, Rect};
+
 #[derive(Default)]
 pub struct HomeScreen {
     worker_info_table: WorkerInfoTable,
     worker_util_table: WorkerUtilTable,
+    worker_info_chart: ClusterOverviewChart,
 }
 
 impl Screen for HomeScreen {
     fn draw(&mut self, frame: &mut DashboardFrame) {
-        let layout = Layout::new(frame);
+        let layout = HomeLayout::new(frame);
         draw_text("HQ top", layout.header_chunk, frame, style_header_text());
         // Update the current selected worker for info table
         let selected_worker = self.worker_util_table.get_selected_item();
         self.worker_info_table.change_info(selected_worker);
 
+        self.worker_info_chart.draw(layout.chart_chunk, frame);
         self.worker_info_table.draw(layout.info_chunk, frame);
         self.worker_util_table.draw(layout.body_chunk, frame);
     }
 
     fn update(&mut self, state: ClusterState) {
-        self.worker_util_table.update(state.overview);
+        self.worker_util_table.update(state.overview.clone());
```

Comment:
I suppose that now is the time to change the parameter of the update method to `&CollectedOverview`.